### PR TITLE
Add HDF5 support for trajs and model_devis

### PR DESCRIPTION
**Pull request by zjgemi**
## Summary by CodeRabbit

- **New Features**
	- Introduced new optional arguments for improved data handling and multitasking capabilities.
	- Added support for HDF5 formatted data in various modules.
	- Enhanced flexibility in input handling for multiple data formats.

- **Bug Fixes**
	- Improved robustness in handling validation data structures.

- **Documentation**
	- Updated documentation to clarify new parameters and their intended use.

### CODE REVIEW

**Review by wanghan-iapcm — COMMENTED** [review]
Could you please also support `run_lmp`, which seems to be straightforward.

File: dpgen2/exploration/selector/conf_selector_frame.py [hunk 1]
```diff
@@ -9,10 +9,14 @@
     List,
     Optional,
     Tuple,
+    Union,
 )
 
 import dpdata
 import numpy as np
```
[the remark this answers is not in the document]
Comment (njzjz) [reply dpgen2/exploration/selector/conf_selector_frame.py:16]: It seems ruff hasn't been enabled in the dpgen2

**Comment by zjgemi** [comment]
> Could you please also support `run_lmp`, which seems to be straightforward.

Sure.

**Comment by zjgemi** [comment]
> Could you please also support `run_lmp`, which seems to be straightforward.

I realize that for `run_lmp`, a task only outputs a single trajectory and a single model_devi file. As outputs of each task must be stored in a seperated file. Merging outputs of each task into a HDF5 file will bring little benefit.

On the other hand, in the HDF5 mode, users cannot conveniently preview file content in UI. That's why HDF5 mode is not employed by default unless performance bottleneck is met.

**Review by wanghan-iapcm — APPROVED** [review]

**Pull request merged**